static/js/components/dashboard/CouponMessage.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+import React from 'react';
+
+import {
+  makeAmountMessage,
+  makeCouponReason,
+} from '../../lib/coupon';
+import {
+  COUPON_CONTENT_TYPE_PROGRAM,
+  COUPON_CONTENT_TYPE_COURSE,
+  COUPON_CONTENT_TYPE_COURSERUN,
+} from '../../constants';
+import type { Coupon } from '../../flow/couponTypes';
+
+export default class CouponMessage extends React.Component {
+  props: {
+    coupon: Coupon,
+  };
+
+  makeCouponTargetMessage = () => {
+    const { coupon } = this.props;
+
+    if (coupon.content_type === COUPON_CONTENT_TYPE_PROGRAM) {
+      return ' for each course in this program';
+    } else if (coupon.content_type === COUPON_CONTENT_TYPE_COURSE ||
+      coupon.content_type === COUPON_CONTENT_TYPE_COURSERUN) {
+      return ' for this course';
+    } else {
+      return '';
+    }
+  };
```

Comment:
I think it might be nice to implement this with a `Map`, mapping coupon types to their corresponding messages. You could declare that at the module level:

```js
const couponTargetMessages = new Map([
  [ COUPON_CONTENT_TYPE_PROGRAM, " for each course in this program" ],
  ...
]);
```

then this method could just look like this:

```js
makeCouponTargetMessage = () => {
  const { coupon } = this.props;
  
  return couponTargetMessages.get(coupon.content_type) || '';
}
```

just a suggestion!